token/client/src/token.rs

```diff
@@ -2201,6 +2201,8 @@ where
         ciphertext_validity_proof_account: Option<&ProofAccount>,
         range_proof_account: Option<&ProofAccount>,
         transfer_amount: u64,
+        transfer_amount_auditor_ciphertext_lo: Option<&PodElGamalCiphertext>,
+        transfer_amount_auditor_ciphertext_hi: Option<&PodElGamalCiphertext>,
```

Comment:
I was having a hard time following when these would be used, so let me know if I've understood. From what I'm seeing, they're only used if all of the equality / ciphertext validity / range proofs are `ProofAccount`s, which makes sense since the function doesn't have access to the ciphertexts otherwise.

In that case, how about creating a new type for the ciphertext validity proof, which is like `ProofAccount`, but also holds onto the auditor ciphertexts? That seems like the clearest option to me, because otherwise this logic is pretty confusing.